models/User.js

```diff
@@ -1,14 +1,25 @@
-const mongoose = require("mongoose");
+const mongoose = require('mongoose');
 const Schema = mongoose.Schema;
 const Role = require("./Role");
 const House = require("./House");
 const Receipt = require("./Receipt");
+const types = require('./types');
 
 const userSchema = new Schema({
   name: String,
-  role: { type: Schema.ObjectId, ref: "Role" },
-  house: { type: Schema.ObjectId, ref: "House" },
-  payment_history: { type: Schema.ObjectId, ref: "Receipt" }
+  role: { 
+    type: String,
+    required: [true, 'Role is not assigned'],
+    enum: types.roles
+  },
+  house: { 
+    type: Schema.ObjectId, 
+    ref: "House", 
+  },
+  payment_history: { 
```

Comment:
Isn't it supposed to be an array? List of receipts like that.. Lol.. Or maybe, we don't need it coz we can make a 'query search' later on when they are looking for their history (?) Any advises? @demounit

Reply:
Yea, I think it should also be an array to allow us to retrieve a list of receipts that can be showed as a ListView for the user. Updated the code to make it an array.